bitnami/discourse/templates/_helpers.tpl

```diff
@@ -120,7 +120,29 @@ imagePullSecrets:
 {{- end -}}
 
 {{/*
-Get the user defined LoadBalancerIP for this release.
+Return the Discourse secret name
+*/}}
+{{- define "discourse.discourseSecretName" -}}
+{{- if not (empty .Values.discourse.existingSecret) }}
```

Comment:
I think `empty` is used for arrays or collections. It may apply to string as well, but I think the standard approach for checking if `.existingSecret` is provided is by simply:

```go
{{- if  .Values.discourse.existingSecret }}
```

Any nil pointer or string whose length is 0 will evaluate to `false`.